atc/db/job.go

```diff
@@ -934,15 +934,15 @@ func (j *job) getRunningBuildsBySerialGroup(tx Tx, serialGroups []string) ([]Bui
 }
 
 func (j *job) getNextPendingBuildBySerialGroup(tx Tx, serialGroups []string) (Build, bool, error) {
-	row := buildsQuery.Options(`DISTINCT ON (b.id)`).
+	row := buildsQuery.
```

Comment:
Given the `buildsQuery`, I can't see the need for `Distinct`

Reply:
@xtreme-sameer-vohra Ya I don't think the distinct was necessary since we are just querying from the builds table. I tested the query out without the distinct and it seems to work properly!

Reply:
Oh now that I look at it again, the distinct might be for if there are multiple rows for a job from the `jobs_serial_groups` table that it joins with. I originally thought it shouldn't need the distinct but I forgot about the fact that jobs can be in multiple serial groups. Thanks for pointing this out!! I will add back the distinct!

Reply:
ah, I didn't realize, the `Options` param would apply to the join on `jobs_serial_groups` as well

Reply:
I think it's because if a job is in multiple serial groups, it will have multiple rows in the `jobs_serial_groups` table and when we query for a build from that job, it will join to the serial groups table combining the build row with all the rows for that job in the `jobs_serial_groups` table (resulting in multiple rows for the same build).